Tools/lint.sh

```diff
@@ -49,9 +66,30 @@ COMMIT=${1:---cached}
 # Get modified files (must be on own line for exit-code handling)
 modified_files=$(git diff --name-only --diff-filter=ACMRTUXB $COMMIT)
 
+function java_check() {
+  echo >&0 "Java changes detected, running Android Studio formatter." || true
+  # expand java_file_list
+  >/dev/null "${ANDROID_STUDIO_ROOT}/bin/format.sh" -s "${JAVA_CODESTYLE_FILE}" -R "${java_temp_dir}"
+
+  # ignore 'added'/'deleted' files, we copied only files of interest to the tmpdir
+  d=$(git diff --diff-filter=ad . "${java_temp_dir}" || true)
+  if ! [ -z "${d}" ]; then
+    echo "!!! Java code is not compliant to coding style, here is the fix:"
+    echo "${d}"
+    fail=1
+  fi
+}
+
 # Loop through each modified file.
 for f in ${modified_files}; do
   # Filter them.
+  if echo "${f}" | egrep -q "[.](java)$"; then
+    # Copy Java files to a temporary directory
+    java_setup
+    mkdir -p $(dirname "${java_temp_dir}"/"${f}")
```

Comment:
Doesn't need 2 pairs on "", can be "${java_temp_dir}/${f}" (same on the next line below)